dialog/file_xdg.go

```diff
@@ -27,7 +28,7 @@ func getFavoriteLocation(homeURI fyne.URI, name, fallbackName string) (fyne.URI,
 	loc = loc[:len(loc)-1]
 	locURI := storage.NewFileURI(string(loc))
 
-	if locURI.String() == homeURI.String() {
+	if strings.TrimRight(locURI.String(), "/") == homeURI.String() {
```

Comment:
Can you be certain the home directory does not include a slash?
It may be better to ensure both are trimmed to cover all cases

Reply:
I have looked into it a bit, and think it could be possible. `homeURI` is constructed from `os.UserHomeDir()`, which will return the value of the environment variable `$HOME` (if set). Usually this shouldn't contain a trailing slash, but if a user manually set their `$HOME` variable and accidentally included the trailing slash, it would be problematic.

Thus I have added another `strings.TrimRight`. It can't hurt.